assets/js/common/DateFilter/DateFilter.jsx

```diff
@@ -0,0 +1,213 @@
+
+
+const oneHour = 60 * 60 * 1000;
+const preconfiguredOptions = {
+  '1h ago': () => new Date(Date.now() - oneHour),
+  '24h ago': () => new Date(Date.now() - 24 * oneHour),
+  '7d ago': () => new Date(Date.now() - 7 * 24 * oneHour),
+  '30d ago': () => new Date(Date.now() - 30 * 24 * oneHour),
+};
+
+const renderOptionItem = (option, placeholder) => {
+  if (!option || !Array.isArray(option)) {
+    return placeholder;
+  }
+  if (typeof option[2] === 'function') {
+    return option[2]();
+  }
+  return option[0];
+};
+
+
+const parseInputOptions = (options) =>
+  options
+    .map((option) => {
+      if (typeof option === 'string' && option in preconfiguredOptions) {
+        return [option, preconfiguredOptions[option]];
+      }
+      if (
+        Array.isArray(option) &&
+        typeof option[1] === 'function' &&
+        option[1]() instanceof Date
+      ) {
+        return option;
+      }
+      return undefined;
+    })
+    .filter(Boolean)
+    .reduceRight(
+      (acc, el) =>
+        acc.find(([label]) => label === el[0]) ? acc : [...acc, el],
+      []
+    )
+    .sort((a, b) => b[1]().getTime() - a[1]().getTime());
+
+const getSelectedOption = (options, value) => {
+  const selectedId = Array.isArray(value) ? value[0] : value;
+  if (typeof selectedId === 'string') {
+    return options.find((option) => option[0] === selectedId);
+  }
+  return undefined;
+};
+
+/**
+ * A component for filtering dates.
+ *
+ * @component
+ * @example
+ * // <DateFilter
+ * //   options={['1h ago', '24h ago', '7d ago', '30d ago']}
+ * //   title="Date"
+ * //   value="24h ago"
+ * //   prefill
+ * //   onChange={(value) => console.log(value)}
+ * // />
+ *
+ * @param {Object} props - The component props.
+ * @param {Array} props.options - The options for the date filter. Each option can be a triple with a an id, a value function and an optional render function.
+ * The value function should return a Date object; the actual date value is calculated at selection time.
+ * In case the render function is not provided, the id will be used as the label.
+ * An option can also be a string, in which case it will be considered as a pre-configured option.
+ * In case more options with the same id are provided, only the last one will be considered.
+ * Options will be displayed sorted by date in descending order.
+ * @param {string} props.title - The title of the date filter, to be shown as placeholder when no value is selected.
+ * @param {string} props.value - The selected id of the selected option. It accepted either a string or an array with the id as the first element.
+ * @param {boolean} props.prefill - Whether to include pre-configured options in the options list. Default is false.
+ * @param {function} props.onChange - The callback function to be called when the value of the date filter changes. It will provide a couple with the selected id and the actual date.
+ */
+function DateFilter({ options = [], title, value, prefill, onChange }) {
```

Comment:
**naming (prefill)** `prefill` or `prefilled`?

Reply:
Done in https://github.com/trento-project/web/pull/2849/commits/f5b9d3d151b9bd1029b59b37d77c17eb6db633fa